src/audio/dai-zephyr.c

```diff
@@ -1554,13 +1554,13 @@ int dai_zephyr_copy(struct dai_data *dd, struct comp_dev *dev, pcm_converter_fun
 	if (dev->direction == SOF_IPC_STREAM_PLAYBACK) {
 		buf_c = buffer_acquire(dd->local_buffer);
 		src_samples = audio_stream_get_avail_samples(&buf_c->stream);
-		sink_samples = free_bytes / sampling;
+		sink_samples = audio_stream_bytes_to_samples(&buf_c->stream, free_bytes);
```

Comment:
personally, I don't suggest have such code, for integer divide, in compiler and its inside lib, there is also a shift operation(its own fast algorithm), it is not complicated, and don't waste too much cycles, however, with new added function and inside if, it do need more cycles.